Fix #28

const-ae opened the pull request:
Unlike most other functions rowLogSumExps first argument is called `lx` instead of `x`. This caused `make_default_method_def("rowLogSumExps")` to generate erroneous code. To fix the issue I inline the correct code manually.

const-ae committed:
Fix #28

Unlike most other functions rowLogSumExps first argument is called `lx`
instead of `x`. This caused `make_default_method_def("rowLogSumExps")`
to generate erroneous code. To fix the issue I inline the correct code
manually.

PeteHaitch commented:
Copying comment from https://github.com/Bioconductor/MatrixGenerics/issues/28#issuecomment-1151795603:

> Looks good to me, but by that logic I think similar fixes also need to be made to some other functions that don't take x as their first argument:
> 
> - `colAvgsPerRowSet()` / `rowAvgsPerRowSet()`

const-ae committed:
Override the [row|col]AvgsPerColSet ANY generics to match the arg names

const-ae committed:
Add test to check that dynamic loading works

const-ae commented:
Hi Pete,

Thanks for the feedback. I have fixed the problem for `[row|col]AvgsPerRowSet()` and added a test that fails on master, but succeeds on this branch. I think the PR is ready to merge :)

Best,
Constantin

PeteHaitch commented:
Thanks, @const-ae.
I made a small change to keep `R CMD check` and have pushed to BioC.
Should be available from `BiocManager::install()` as `v1.9.1` in 1-2 days.
I will also backport this to the current release branch.
